antidotum/tergo/src/rust/src/lib.rs

```diff
@@ -72,7 +72,17 @@ fn format_code(source_code: &str, configuration: extendr_api::List) -> String {
             .unwrap_or(default_config.insert_newline_in_quote_call.0),
     );
 
-    tergo_lib::tergo_format(source_code, Some(&config)).unwrap()
+
+    let result = match tergo_lib::tergo_format(source_code, Some(&config)) {
+        Ok(formatted_code) => {
+            list!("success", formatted_code)
+        }
+        Err(error) => {
+            list!("error", error)
+        }
+    };
+    
+    return result;
```

Comment:
```suggestion
    match tergo_lib::tergo_format(source_code, Some(&config)) {
        Ok(formatted_code) => {
            list!("success", formatted_code)
        }
        Err(error) => {
            list!("error", error)
        }
    }
```
A bit more idiomatic for rust. This works. There's even an entry in the docs of extend https://docs.rs/extendr-api/latest/extendr_api/#returning-resultt-e-to-r about doing something similar.